src/app/ReadHandler.cpp

```diff
@@ -56,6 +60,15 @@ CHIP_ERROR ReadHandler::Init(Messaging::ExchangeManager * apExchangeMgr, Interac
 
 void ReadHandler::Shutdown(ShutdownOptions aOptions)
 {
+    if (IsSubscription())
+    {
+        mSubscriptionId     = 0;
```

Comment:
I suggest to move everything except CancelTimer outside of the if: it is easier to reason about saying  'shutdown ensures these values are 0'.